docs/remote-k8s-access/README.md

```diff
@@ -0,0 +1,179 @@
+
+The secret should have a `kubeconfig` field that containts the contents of the file. 
+
+Here is an example:
+
+```yaml
+apiVersion: v1
+data:
+  kubeconfig: ... 
+kind: Secret
+metadata:
+  creationTimestamp: "2021-09-23T18:58:24Z"
+  name: kind-k8ssandra-0-config
+  namespace: default
+  resourceVersion: "164744"
+  uid: 722fff76-8836-4360-849f-ea55d610a26c
+type: Opaque
+```
+
+# ClientConfig
+K8ssandra Operator uses a kubeconfig file to create a remote client connection. A service account provides authentication and authorization with the api server. The operator accesses the kubeconfig file via a secret.
+
+A ClientConfig is a reference to a kubeconfig secret. At startup the operator queries for all ClientConfigs in the namespace it is configured to watch. It iterates through the ClientConfigs and creating a remote client for each one. 
+
+a kubeconfile can contain access to multiple clusters. The operator only creates a remote client for the kube context specified by the ClientConfig.
+
+The operator stores the remote clients in a cache that persists for the lifetime of the operator and is used across all K8ssandraClusters. 
+
+Let's look at an example ClientConfig:
+
+```yaml
+apiVersion: k8ssandra.io/v1alpha1
+kind: ClientConfig
+meta:
+  name: kind-k8ssandra-1
+spec:
+  contextName: kind-k8ssandra-1
+  kubeConfigSecret: 
+    name: kind-k8ssandra-1-config
+```
+
+The `contextName` field is optional. If omitted the operator will look for a kube context having the same name as the ClientConfig.
+
+## Using a ClientConfig
+Suppose we have two additional ClientConfigs similar to the one in the previous example - `kind-k8ssandra-2` and `kind-k8ssandra-3`. We want to create a K8ssandraCluster with three datacenters, one per Kubernetes cluster.
+
+The K8ssandraCluster manifest would look like this:
+
+```yaml
+apiVersion: k8ssandra.io/v1alpha1
+kind: K8ssandraCluster
+metadata:
+  name: demo
+spec:
+  cassandra:
+    cluster: demo
+    serverVersion: "4.0.1"
+    storageConfig:
+      cassandraDataVolumeClaimSpec:
+        storageClassName: standard
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: 5Gi
+    datacenters:
+      - metadata:
+          name: dc1
+        k8sContext: kind-k8ssandra-1
+        size: 3
+      - metadata:
+          name: dc2
+        k8sContext: kind-k8ssandra-2
+        size: 3
+      - metadata:
+          name: dc3
+        k8sContext: kind-k8ssandra-3
+        size: 3
+```
+
+Notice the `k8sContext` property of each element in the `datacenters` array. It should specify the name of a kube context that was provided by a ClientConfig. 
+
+The operator will use the value of `k8sContext` to look up the remote client from the cache and then use it to create the CassandraDatacenter in the remote cluster.
+
+As we can see, the ClientConfig is not used by the K8ssandraCluster. It only needs to know the kube context names.
```

Comment:
It just occurred to me: couldn't we get rid of ClientConfig and put the secret name directly here? E.g.

```
    datacenters:
      - metadata:
          name: dc1
        k8sContext: kind-k8ssandra-1
        kubeconfigSecret: kind-k8ssandra-1-kubeconfig-secret
        size: 3
      - metadata:
          name: dc2
        k8sContext: kind-k8ssandra-2
        kubeconfigSecret: kind-k8ssandra-2-kubeconfig-secret
        size: 3
      - metadata:
          name: dc3
        k8sContext: kind-k8ssandra-3
        kubeconfigSecret: kind-k8ssandra-3-kubeconfig-secret
        size: 3
```